src/components/Card/index.jsx

```diff
@@ -1,27 +1,88 @@
 import React, { Component } from 'react';
 import './index.css';
 
+import {BsPencil, BsX, BsCheck} from 'react-icons/bs';
+
 class Card extends Component {
+
     state = {
-        isChecked: false
+        isChecked: false,
+        isEdit: false,
+        cardData : {
+            cardLogo: 'Caption',
+            text: 'Some text here...',
+        },
+        tempCardData: {
+            tempLogo: 'Caption',
+            tempText: 'Some text here...'
+        }
     }
+    
 
     changeStyles = () => {
         this.setState({isChecked : !this.state.isChecked});
     }
+    changeEditStatus = () => {
+        this.setState({isEdit : !this.state.isEdit, isChecked: false});
+    }
+    handleChangeLogo = (event) => {
+        console.log('piu');
+        this.setState({tempCardData: {tempLogo: event.target.value, tempText: this.state.tempCardData.tempText}});
+    }
+    handleChangeText = (event) => {
```

Comment:
Since handleChangeText and handleChangeLogo are making the same thing - make this function common and reusable: pass property name as function argument and assign it as dynamic property. In case of 2 inputs - no so much difference, in case of 100 - you'll have one handler function instead of 100.